source/LibMultiSense/details/channel.cc

```diff
@@ -705,11 +705,17 @@ void *impl::statusThread(void *userDataP)
                 const utility::TimeStamp latency((pong.getNanoSeconds() - ping.getNanoSeconds()) / 2);
 
                 //
-                // Compute and apply the estimated time offset
+                // If it took less than 5ms each direction for transmission consider this for a valid time offset
 
-                const utility::TimeStamp offset = ping + latency - msg.uptime;
+                if (latency.getNanoSeconds() < 5000000) {
```

Comment:
Just for readability
```suggestion
                if (latency.getNanoSeconds() < 5'000'000) {
```